azavea/pfb-network-connectivity
# Add AnalysisLocalUploadTask model + serializers + views

## Overview

Adds model, serializers and views for AnalysisLocalUploadTask. 

Custom create serializer so that we can send the
neighborhood UUID. Further input validation and/or
kickoff of jobs should happen in serializer validate
method or model save as appropriate.

View supports POST, LIST, GET but not PUT or PATCH since
these objects are effectively immutable from the user's
perspective once created.

### Demo

![screen shot 2018-12-10 at 4 51 30 pm](https://user-images.githubusercontent.com/1818302/49763900-ec562880-fc9b-11e8-8728-f2f3fde2f483.png)

### Notes

I went with the custom create serializer to clearly encapsulate the logic necessary to create a new instance of this model and to show that the neighborhood gets translated to a new AnalysisJob instance that the task model will use. Alternatively, we could have kept this POST simple by taking a direct AnalysisJob UUID, generated by first POSTing to /api/analysis-jobs/ but for this workflow we don't want that because we don't want to immediately kick off a new AnalysisJob run for the model which is what that endpoint does.

## Testing Instructions

`./scripts/django-manage migrate` to run migrations, then hit http://localhost:9202/api/local_upload_tasks/. You should be able to create a new model object via the DRF browser view or by direct curl or httpie requests. Note that a new AnalysisJob object is created for the task once the 201 CREATED is returned.

Closes #619
